qcfractal/interface/collections/dataset.py

```diff
@@ -622,20 +639,41 @@ def _get_molecules(self, indexer: Dict[str, 'ObjectId']) -> 'pd.Series':
             If no records match the query
         """
 
-        molecules = []
         molecule_ids = list(set(indexer.values()))
-        for i in range(0, len(molecule_ids), self.client.query_limit):
-            molecules.extend(self.client.query_molecules(id=molecule_ids[i:i + self.client.query_limit]))
 
-        molecules = pd.DataFrame.from_dict([{"molecule_id": x.id, "molecule": x} for x in molecules])
-        if len(molecules) == 0:
+        # Partition request into cached and uncached molecules
+        cached_ids = []
+        cached_molecules = []
+        uncached_ids = []
+        uncached_molecules = []
+        for molecule_id in molecule_ids:
```

Comment:
It might be better to cache at the client level. Where for all queries that include `id` we cache say 100k of them. This would be more universal and  would make all of this simpler.

Reply:
This depends on whether we want an in-memory cache for molecules loaded from a `.download()`-ed HDF5 file or just for network-retrieved molecules.